pynsee/geodata/geofrdataframe.py

```diff
@@ -0,0 +1,122 @@
+import warnings
+from typing import Union
+
+import numpy as np
+from pandas import DataFrame
+from geopandas import GeoDataFrame, GeoSeries
+
+
+class GeoFrDataFrame(GeoDataFrame):
+    """
+    Class for handling GeoDataFrames built from IGN's geographical data.
+    It inherits from :class:`~geopandas.GeoDataFrame`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def _constructor(self):
+        return GeoFrDataFrame
+
+    def _constructor_from_mgr(
+        self, mgr, axes
+    ) -> Union[DataFrame, "GeoFrDataFrame"]:
+        """Patch to avoid class change"""
+        obj = super()._constructor_from_mgr(mgr, axes)
+
+        if isinstance(obj, GeoDataFrame):
+            obj.__class__ = GeoFrDataFrame
+
+        return obj
+
+    def __getitem__(self, key):
+        """
+        Patch to avoid class change.
+        """
+        result = super().__getitem__(key)
+
+        if isinstance(result, GeoDataFrame):
+            result.__class__ = GeoFrDataFrame
+
+        return result
+
+    def copy(self, deep: bool = True) -> "GeoFrDataFrame":
+        """Patch to avoid class change"""
+        copied = super().copy(deep=deep)
+        copied.__class__ = GeoFrDataFrame
+        return copied
+
+    def get_geom(self):
+        """
+        Deprecated alias for `geometry`.
+
+        .. deprecated:: 0.2.0
+
+            Use :attr:`~geopandas.GeoDataFrame.geometry` instead.
```

Comment:
Use :attr:`~GeoFrDataFrame.geometry` instead ?

Reply:
no, after the documentation overhaul I'm preparing, this will link directly to geopandas' doc